### DISCUSSION TITLE
Error when inherit "ForeignKey" column from super class.

### DISCUSSION THREAD

**Discussion by Yushihu** [fastapi/sqlmodel#938] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from datetime import datetime, timezone
from json import dumps, loads
from typing_extensions import Unpack
from sqlmodel import Field, Relationship, Session, SQLModel, create_engine, select, ForeignKey
from sqlalchemy.engine import Engine
from sqlalchemy.orm import Session
from sqlalchemy import Integer, UniqueConstraint, event, Column, String, TypeDecorator, JSON, ForeignKey
from pydantic import BaseModel, ConfigDict
from enum import Enum


class Colors(Enum):
    RED = "red"
    BLUE = "blue"
    YELLOW = "yellow"


def _mod(cls: type[BaseModel]):
    class MyType(TypeDecorator):
        '''Prefixes Unicode values with "PREFIX:" on the way in and
        strips it off on the way out.
        '''

        impl = JSON

        cache_ok = True

        def process_bind_param(self, value: BaseModel, dialect):
            if value is None:
                return None
            return value.model_dump()

        def process_result_value(self, value, dialect):
            if value is None:
                return None
            return cls(**value)

    return MyType


@event.listens_for(Engine, "connect")
def set_sqlite_pragma(dbapi_connection, connection_record):
    cursor = dbapi_connection.cursor()
    cursor.execute("PRAGMA foreign_keys=ON")
    cursor.close()


class Team(SQLModel, table=True):
    id: int | None = Field(default=None, sa_column=Column("Noo", Integer, primary_key=True))
    name: str = Field(index=True, max_length=123)
    headquarters: str

    heroes: list["Hero"] = Relationship(back_populates="team")


class Hero(SQLModel, table=True):
    id: int | None = Field(default=None, primary_key=True)
    name: str = Field(index=True)
    secret_name: str
    age: int | None = Field(default=None, index=True)

    team_id: int | None = Field(default=None, foreign_key="team.Noo")
    team: Team | None = Relationship(back_populates="heroes")


class HeroItem(SQLModel):
    hero_id: int = Field(sa_type=ForeignKey(Hero.id, ondelete="CASCADE"), primary_key=True)


class Mask(HeroItem, table=True):
    pass


class Suit(HeroItem, table=True):
    pass
```

### Description

The Mask and Suit share the "ForeignKey" instance which cause the error below.
Could we have "sa_column_factory" in parameters of "Field" which i thought a more general solution to mix-in sqlalchemy Column with sqlmodel. Then the solution of the problem could be.
`
def refer_to_hero_id():
    return Column(ForeignKey(Hero.id, ondelete="CASCADE"), primary_key=True)

class HeroItem(SQLModel):
    hero_id: int = Field(sa_column_factory=refer_to_hero_id)

`
`
  File "G:\test\venv\Lib\site-packages\sqlmodel\main.py", line 474, in __new__
    col = get_column_from_field(v)
          ^^^^^^^^^^^^^^^^^^^^^^^^
  File "G:\test\venv\Lib\site-packages\sqlmodel\main.py", line 661, in get_column_from_field
    return Column(sa_type, *args, **kwargs)  # type: ignore
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "G:\test\venv\Lib\site-packages\sqlalchemy\sql\schema.py", line 2145, in __init__
    self._init_items(*cast(_typing_Sequence[SchemaItem], l_args))
  File "G:\test\venv\Lib\site-packages\sqlalchemy\sql\schema.py", line 231, in _init_items
    spwd(self, **kw)
  File "G:\test\venv\Lib\site-packages\sqlalchemy\sql\base.py", line 1324, in _set_parent_with_dispatch
    self._set_parent(parent, **kw)
  File "G:\test\venv\Lib\site-packages\sqlalchemy\sql\schema.py", line 3213, in _set_parent
    raise exc.InvalidRequestError(
sqlalchemy.exc.InvalidRequestError: This ForeignKey already has a parent !
`

### Operating System

Windows

### SQLModel Version

0.0.16

### Python Version

3.11

**Comment by Yushihu** [comment]
Sorry about the format.
```
def refer_to_hero_id():
    return Column(ForeignKey(Hero.id, ondelete="CASCADE"), primary_key=True)


class HeroItem(SQLModel):
    hero_id: int = Field(sa_column_factory=refer_to_hero_id)

```

**Comment by YuriiMotov** [comment]
```
    hero_id: int = Field(sa_type=ForeignKey(Hero.id, ondelete="CASCADE"), primary_key=True)
```
This is not how foreign key is supposed to be defined.
Please, read docs:
* https://sqlmodel.tiangolo.com/tutorial/relationship-attributes/
* https://sqlmodel.tiangolo.com/tutorial/relationship-attributes/cascade-delete-relationships/
